src/App/App.js

```diff
@@ -1,22 +1,46 @@
 import React, { Component } from 'react';
-import { Route, NavLink } from 'react-router-dom';
+import { Route, NavLink, withRouter } from 'react-router-dom';
 import './App.css';
 import LoginForm from '../LoginForm/LoginForm';
 import Areas from '../Areas/Areas';
 
 class App extends Component {
   constructor() {
     super();
+    this.state = {
+      name: '',
+      email: '',
+      type: ''
+    }
+  }
+
+  change = (key, event) => {
+    this.setState({ [key]: event.target.value })
+  }
+
+  submit = () => {
+    if (this.state.name === '' || this.state.email === '' || this.state.type === '') {
+      document.querySelector('.error').classList.remove('hidden');
+    } else {
+      this.props.history.push('/areas')
+    }
   }
 
   render() {
     return (
       <main>
-        <Route path='/' exact component={ LoginForm } />
-        <Route path='/areas' exact component={ Areas } />
+        <Route
+        path='/' exact render={
+          (props) => <LoginForm change={this.change} submit={this.submit}/>
+        }/>
+
+        <Route
+        path='/areas' exact render={
+          (props) => <Areas name={this.state.name} type={this.state.type}/>
+        }/>
       </main>
     )
   }
 }
 
-export default App;
+export default withRouter(App);
```

Comment:
I'm not sure I follow why we need withRouter on App. Can you elaborate on what exactly this is doing that is different from before? My reading shows that withRouter gives access to use props.history.push, which was already in use before this PR, so I don't think I fully understand the full functionality of withRouter.

Reply:
Yeah, I don't 100% understand why it worked without this when in the login component but requires it not that it's in the app component. I think it's because the login form component is already wrapped in a route, which gives it access to this.props.history, but app wasn't wrapped in anything and so didn't have access to it. My understanding is that wrapping it in withRouter gives it access to this.props.history (again, still a bit fuzzy on this and hoping it clears up as we get more practice)

Reply:
Ok, it's still fuzzy for me too, but I think I see what you're saying. I will merge the PR and work off this new version- I think playing with the code will make the 'how' a little less fuzzy. Thanks!